yapapi/props/builder.py

```diff
@@ -87,3 +79,21 @@ def add(self, m: Model):
     async def subscribe(self, market: Market) -> Subscription:
         """Create a Demand on the market and subscribe to Offers that will match that Demand."""
         return await market.subscribe(self._properties, self.constraints)
+
+    async def decorate(self, *decorators: 'DemandDecorator'):
+        for decorator in decorators:
+            await decorator.decorate_demand(self)
+
+
+class DemandDecorator(abc.ABC):
+    """An interface that specifies classes that can add properties and constraints through a DemandBuilder"""
+
+    @abc.abstractmethod
+    async def decorate_demand(self, demand: DemandBuilder):
```

Comment:
It's a very minor comment, but I'm not convinced by this "decorating" terminology (`DemandDecorator` / `decorate_demand`)? To me "decorating" sounds like we're adding some "ornaments" to a Demand while in fact we're constructing it.

I know that those `decorate_demand` methods were there already. And I don't have a better name in mind at the moment.

Reply:
well, if you think about it, you get a plain demand and then, what we do is add additional pieces to it ... so, you _could_ say we're incrementally decorating it with "ornaments" to arrive at a final, constructed demand ... 

whether this is good terminology, I'm not fully convinced either... but if we'd like not to explode the "renaming stuff" activity beyond reasonable minimum, I'd stay with `decorate` et al